contract/src/test/kotlin/io/provenance/scope/loan/contracts/RecordENoteContractUnitTest.kt

```diff
@@ -1,50 +1,234 @@
+                            newENote.controller.isSet() shouldBe true
+                            newENote.signedDate.isSet() shouldBe true
+                            newENote.vaultName.isNotBlank() shouldBe true
+                        }
+                    }
+                }
             }
         }
-        "given a valid input to an empty scope" xshould {
-            "not throw an exception" {
-                // TODO: Implement
+        "recordServicingData for an empty scope" When {
+            "given an empty input" should {
+                "throw an appropriate exception" {
+                    shouldThrow<ContractViolationException> {
+                        recordServicingData(ServicingData.getDefaultInstance())
+                    }.let { exception ->
+                        exception shouldHaveViolationCount 1U
+                        exception.message shouldContain "Servicing data is not set"
+                    }
+                }
             }
-        }
-        "given a valid input to update the existing servicing data record" xshould {
-            "not throw an exception" {
-                // TODO: Implement
+            "given an input with an invalid document" should {
+                "throw an appropriate exception" {
+                    checkAll(
+                        loanStateSet(size = 1),
+                        anyValidDocumentMetadata,
+                        anyValidDocumentMetadata,
+                        anyInvalidUuid,
+                    ) { randomLoanState, randomFirstValidDocument, randomSecondDocument, randomInvalidId ->
+                        shouldThrow<ContractViolationException> {
+                            recordServicingData(
+                                ServicingData.newBuilder().also { servicingDataBuilder ->
+                                    servicingDataBuilder.addAllLoanState(randomLoanState)
+                                    servicingDataBuilder.clearDocMeta()
+                                    servicingDataBuilder.addDocMeta(randomFirstValidDocument)
+                                    servicingDataBuilder.addDocMeta(
+                                        randomSecondDocument.toBuilder().also { invalidDocumentBuilder ->
+                                            invalidDocumentBuilder.id = randomInvalidId
+                                        }.build()
+                                    )
+                                }.build()
+                            )
+                        }.let { exception ->
+                            exception shouldHaveViolationCount 1U
+                            exception.message shouldContain "Document must have valid ID"
+                        }
+                    }
+                }
+            }
+            "given an invalid input" xshould {
+                "throw an appropriate exception" {
+                    // TODO: Implement
```

Comment:
Is this a leftover or soon to be something more specific?

Reply:
Good catch, it's a leftover from breaking out "given an invalid input" cases into specific variants